src/composables/useTooltips.ts

```diff
@@ -0,0 +1,75 @@
+import { getFeatureId } from "../resources/utils";
+import { APP_BASE_ROUTE } from "../router/utils";
+
+const router = useRouter();
+
+export const useTooltips = (
+    appConfig: Ref<AppConfig>,
+    featureRefs: Ref<(typeof LGeoJson)[]>,
+    selectedIndicator: Ref<string>,
+    selectedIndicators: Ref<Dict<FeatureIndicatorValues>>,
+    selectedCountryId: Ref<string>
+) => {
+    const tooltipForFeature = (feature: Feature) => {
+        let indicatorValues = "";
+        const featureId = getFeatureId(feature, appConfig.value);
+        if (featureId in selectedIndicators.value) {
+            const featureValues = selectedIndicators.value[featureId];
+            indicatorValues = Object.keys(featureValues)
+                .sort((key) => (key === selectedIndicator.value ? -1 : 1))
+                .map((key) => {
+                    const { humanReadableName, unit } = appConfig.value.indicators[key];
+                    const { mean } = featureValues[key];
+                    const headlineNumber = mean.toPrecision(3);
+                    const line = `${humanReadableName}: ${headlineNumber} ${unit}<br/>`;
+                    return key === selectedIndicator.value ? `<span class="font-weight-bold">${line}</span>` : line;
+                })
+                .join("");
+        }
+        const name = feature.properties![appConfig.value?.featureNameProp] || featureId;
+        return `<div class="text-body-1">${name}</div>` + `<div class="text-body-2">${indicatorValues}</div>`;
+    };
+
+    const createTooltips = {
+        onEachFeature: (feature: Feature, layer: Layer) => {
+            layer.bindTooltip(tooltipForFeature(feature)).openTooltip();
+            layer.on({
+                click: async () => {
+                    const country = feature.properties[appConfig.value?.featureCountryProp];
+                    // select feature's country, or unselect if click on it when already selected
+                    let countryToSelect: string;
+                    if (country === selectedCountryId.value) {
+                        countryToSelect = "";
+                    } else {
+                        countryToSelect = country;
+                    }
+                    router.push(`/${APP_BASE_ROUTE}/${selectedIndicator.value}/${countryToSelect}`);
+                }
+            });
+        }
+    };
+
+    const updateTooltips = (featuresWithColours: FeatureWithColour[]) => {
+        featuresWithColours.forEach((f: FeatureWithColour) => {
+            const geojson = featureRefs.value.find((fr) => {
+                const frId = getFeatureId(fr.geojson, appConfig.value);
+                return frId === f.id;
+            });
+            if (geojson && geojson.geojson && geojson.leafletObject) {
+                geojson.leafletObject.eachLayer((layer: Layer) => {
+                    layer.setTooltipContent(tooltipForFeature(f.feature));
+                });
+            }
+        });
+    };
```

Comment:
I've refactored this a bit for performance in the full data branch as well - `featuresWithColours` is now a dictionary rather than than an array, for faster findabiliity. We now loop through the refs and find the related FeatureWithColour. 
https://github.com/mrc-ide/arbomap/blob/715919a79b9a7e08ba0cebad6f2e4f7de4e52ad7/src/components/Choropleth.vue#L154